Rules/ReviewUnusedParameter.cs

```diff
@@ -36,6 +36,12 @@ public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
 
             foreach (ScriptBlockAst scriptBlockAst in scriptBlockAsts)
             {
+                // bail out if PS bound parameter used.
+                if (scriptBlockAst.Find(IsBoundParametersReference, false) != null)

```

Comment:
Using `.Any()` we can bail out on the first find:
```suggestion
                if (scriptBlockAst.Find(IsBoundParametersReference, false).Any())

```

Reply:
I thought` .Any()` is not required with [Find()](https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.language.ast.find?view=powershellsdk-7.0.0#System_Management_Automation_Language_Ast_Find_System_Func_System_Management_Automation_Language_Ast_System_Boolean__System_Boolean_) used, becuase it is going to return the first node for which predicate returns true. Let me know if I still have to use `.Any()`

Reply:
You're right, sorry. Because PSSA has to frequently use the `FindAll()` method, I didn't notice you were actually using `.Find()`, which I agree is the better choice here :-)